libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2354,3 +2354,49 @@ std::string YulUtilFunctions::tryDecodeErrorMessageFunction()
 		).render();
 	});
 }
+
+string YulUtilFunctions::returndatacopyFunction()
+{
+	string const functionName = "returndata_copy";
+
+	if (!m_evmVersion.supportsReturndata())
+	{
+		return util::Whiskers(R"(
+			function <functionName>() -> data {
+				data := <zeroPointer>
+			}
+		)")
+		("functionName", functionName)
+		("zeroPointer", to_string(CompilerUtils::zeroPointer))
+		.render();
+	}
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return util::Whiskers(R"(
+			function <functionName>() -> data {
+				switch returndatasize()
+				case 0 {
+					data := 0x60
+				}
+				default {
+					// NB: We could maybe use allocateMemoryArrayFunction(...) for allocating
+					// our data here, too?
+
+					// allocate some memory into data of size returndatasize() + PADDING
+					data := mload(<freeMemoryPointer>)
```

Comment:
Please at least use the allocation function.

Reply:
`m_utils.allocationFunction()` - oh nice, didn't know that one exists. Thanks!